lib/matplotlib/axes/_axes.py

```diff
@@ -3318,8 +3318,11 @@ def get_next_color():
                     raise TypeError(
                         'autopct must be callable or a format string')
                 
-                if mpl.rcParams["text.usetex"] or 'usetex' in textprops:
-                    s = s.replace("%", "\%")
+                if 'usetex' in textprops:
+                    if textprops['usetex']:
+                        s = s.replace("%", "\%")
+                elif mpl.rcParams["text.usetex"]:
+                    s = s.replace("%", r"\%")
```

Comment:
```suggestion
                if textprops.get("usetex", mpl.rcParams["text.usetex"]):
                    s = s.replace("%", r"\%")
```

Reply:
Thanks.
I think this is much cleaner way to write the code.